### DISCUSSION TITLE
About v0.5

### DISCUSSION THREAD

**Discussion by ImSingee** [rwf2/Rocket#2538] [General]
I really love Rocket. And I (and many community members may) want to know the Rocket's current state.

When will the next rc version of 0.5 be launched? And when will the stable version of 0.5 be launched? What's blocked the release of rocket v0.5? Is there anything I (or other community members) can help to boost the release?

Politely @SergioBenitez

**Comment by SergioBenitez** [comment]
Hi @ImSingee, thanks for bringing this up.

> And I (and many community members may) want to know the Rocket's current state.

I'm still actively maintaining Rocket, though the "I" is something I'd like to change sooner than later. Not in the sense that I will no longer be maintaining Rocket (I love Rocket, too!), but in the sense that I'd like to increase the number of maintainers and external contributions.

In that vein, I've been preparing the launch of [RWF2](https://rwf2.org), a foundation to support the project, financially and developmentally. One of the foundation's core tasks, in addition to funding, will be making apparent and clear the process to contributing and eventually becoming a maintainer. I'm still working on the specifics, and an unfortunate incident resulted in a major set-back, but rest assured that I'll be discussing more about RWF2 soon.


> When will the next rc version of 0.5 be launched? And when will the stable version of 0.5 be launched?

There will be no further release candidates for 0.5. The `v0.5` branch as it stands now, with at most minor modifications, will be the GM for 0.5. I'm actively working on preparing the release, and I expect that it will happen within the next two weeks.

> What's blocked the release of rocket v0.5?

Some of it has been personal (completely unforeseeable life events), some of it has been professional (I've been working on some awesome projects with a commercial intent that I'm excited to announce later this year), and some of it has been a failure of project management. Rocket v0.5 has been ready for quite some time. In earnest, each release candidate could have served as a major release - the polish, both architecturally and documentation wise, was certainly there, especially considered in context. But, for one reason or another, I've justified holding back the release, much to the project's detriment. I'm hoping that RWF2 will also help in this regard by codifying release conventions that ultimately prevent a repeat of these events.

> Is there anything I (or other community members) can help to boost the release?

Nothing at the moment. This is yet another aspect that requires great attention (enabling contributions to the project in ways other than pull requests), and yet another duty I'd like to ascribe to the foundation.

**Comment by ImSingee** [comment]
@SergioBenitez Thanks for your reply! I really love Rocket and If it falls, it would be a loss for the entire Rust community. Hope for v0.5 and hope for RWF2!
